admin/js/block-editor-custom-alignments-admin.js

```diff
@@ -353,13 +354,18 @@
 								} );
 
 								updatedClass = classArray.join( ' ' );
-							} else {
+							} else if ( slug !== false ) {
 								updatedClass = `align${ slug }`;
+							} else {
+								updatedClass = '';
 							}
 
 							// Change the block align attribute
 							props.setAttributes( {
-								align: slug === undefined ? 'none' : slug,
+								align:
+									slug === undefined || slug === false
+										? 'none'
+										: slug,
```

Comment:
Hmm, good call. After you called this out I realized I was missing some checks anyway. I reworked the `onclick` function a bit. Removing the `false` override (it wasn't useful) and instead trading it for a simple `if` & `replace`/`trim` for removing `alignnone` if necessary.

I'm wondering though, would you have a more simple way to manage a class list that relies so heavily on whether or not one class is set in the list? Is there a way I can reduce the if chain?

Reply:
Looking at this closer, I'm curious why we're opting to remove the `alignnone` at all? As I understand it, that css class is part of core WP, correct? Even if we don't plan to style for it, we can leave it as-is and not worry about conflicts with any core changes in the future, correct?

Reply:
`alignnone` doesn't do anything in WP. When you select the "None" alignment in WP the target selectors use `:not(.alignleft, .alignright, .alignwide)` in order to style the content width. So yes, I suppose we could leave it in, but that would be against the default WP implementation. I can see where it could be helpful though if we wanted to leave it. I'll leave it up to you. It wouldn't simplify the `if` statements at all but we could at least get rid of the last check for if `alignnone` exists.